test/unix/test_lwt_bytes.ml

```diff
@@ -732,55 +739,76 @@ let suite = suite "lwt_bytes" [
 
     test "mincore buffer length = page_size * 2, n_states = 1"
       ~only_if:(fun () -> not Sys.win32) begin fun () ->
-      test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 1
-      >>= fun states ->
-      Lwt.return (states.(0) = true)
+      Lwt.catch
+        (fun () ->
+           test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 1
+           >>= fun () -> Lwt.return true
+        )
+        (function
+          | Invalid_argument _message -> Lwt.return false
+          | _ -> Lwt.return false
+        )
     end;
 
     test "mincore buffer length = page_size * 2, n_states = 2"
       ~only_if:(fun () -> not Sys.win32) begin fun () ->
       Lwt.catch
         (fun () ->
-        test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 2
-        >>= fun states ->
-        Lwt.return (states.(0) = false)
+           test_mincore (Lwt_bytes.page_size * 2) Lwt_bytes.page_size 2
+           >>= fun () -> Lwt.return false
         )
         (function
-      | Invalid_argument _message -> Lwt.return true
-      | _ -> Lwt.return false
+          | Invalid_argument _message -> Lwt.return true
+          | _ -> Lwt.return false
```

Comment:
Here, it's better to reject the promise, so the testing framework reports which exception we actually got:

`| exn -> Lwt.fail exn`